host-configs/llnl/toss_4_x86_64_ib/intel@2022.1.0-c++17.cmake

```diff
@@ -13,10 +13,10 @@
 #------------------------------------------------------------------------------
 
 #------------------------------------------------------------------------------
-# intel-oneapi.2021.3 compilers
+# intel-2022.1.0 compilers
 #------------------------------------------------------------------------------
 
-set(INTELLLVM_VERSION "intel-oneapi.2021.3")
+set(INTELLLVM_VERSION "intel-2022.1.0-magic")
```

Comment:
have we ever figured out what "magic" means?

Reply:
nope.

Reply:
From https://hpc.llnl.gov/software/development-environment-software/compilers-lc

Magic compilers: a "wrapped" version of the compiler, to make it more robust for users in LC environments. This "wrapped" version is what LC has traditionally deployed (seen on TOSS 3 and CORAL systems). This form of the compilers are best for complex build process and always ensure that resulting binaries do not depend on the state of a user's environment. They also ensure that executables can be run across similar LC platforms. These compilers are accessed using LMod and have the naming convention vendor/version-magic.

I think it has something to do with setting RPATHS who knows.